packages/contracts/test/asset_proxy_dispatcher/dispatcher.ts

```diff
@@ -0,0 +1,330 @@
+        });
+    });
+
+    describe('getAssetProxy', () => {
+        it('should return correct address of registered proxy', async () => {
+            await assetProxyDispatcher.addAssetProxy.sendTransactionAsync(
+                AssetProxyId.ERC20,
+                erc20TransferProxy.address,
+                ZeroEx.NULL_ADDRESS,
+                { from: owner },
+            );
+            const proxyAddress = await assetProxyDispatcher.getAssetProxy.callAsync(AssetProxyId.ERC20);
+            expect(proxyAddress).to.be.equal(erc20TransferProxy.address);
+        });
+
+        it('should return NULL address if requesting non-existent proxy', async () => {
+            const proxyAddress = await assetProxyDispatcher.getAssetProxy.callAsync(AssetProxyId.ERC20);
+            expect(proxyAddress).to.be.equal(ZeroEx.NULL_ADDRESS);
+        });
+    });
+
+    describe('transferFrom', () => {
+        it('should dispatch transfer to registered proxy', async () => {
+            // Register ERC20 proxy
+            await assetProxyDispatcher.addAssetProxy.sendTransactionAsync(
+                AssetProxyId.ERC20,
+                erc20TransferProxy.address,
+                ZeroEx.NULL_ADDRESS,
+                { from: owner },
+            );
+            // Construct metadata for ERC20 proxy
+            const encodedProxyMetadata = encodeERC20ProxyData(zrx.address);
+            // Perform a transfer from makerAddress to takerAddress
+            const balances = await dmyBalances.getAsync();
+            const amount = new BigNumber(10);
+            await assetProxyDispatcher.transferFrom.sendTransactionAsync(
+                encodedProxyMetadata,
+                makerAddress,
+                takerAddress,
+                amount,
+                { from: exchangeAddress },
+            );
+            // Verify transfer was successful
+            const newBalances = await dmyBalances.getAsync();
+            expect(newBalances[makerAddress][zrx.address]).to.be.bignumber.equal(
+                balances[makerAddress][zrx.address].minus(amount),
+            );
+            expect(newBalances[takerAddress][zrx.address]).to.be.bignumber.equal(
+                balances[takerAddress][zrx.address].add(amount),
+            );
+        });
+
+        it('should throw if delegating to unregistered proxy', async () => {
+            // Construct metadata for ERC20 proxy
+            const encodedProxyMetadata = encodeERC20ProxyData(zrx.address);
+            // Perform a transfer from makerAddress to takerAddress
+            const balances = await dmyBalances.getAsync();
+            const amount = new BigNumber(10);
+            return expect(
+                assetProxyDispatcher.transferFrom.sendTransactionAsync(
+                    encodedProxyMetadata,
+                    makerAddress,
+                    takerAddress,
+                    amount,
+                    { from: notOwner },
```

Comment:
This should be `{ from: exchangeAddress }` to make sure the tx is failing for the right reason.